curvefs/src/client/mds_client.cpp

```diff
@@ -17,37 +17,226 @@
+            LOG(WARNING) << "DeleteFs Failed, errorcode = " << cntl->ErrorCode()
+                         << ", error content:" << cntl->ErrorText()
+                         << ", log id = " << cntl->log_id();
+            return -cntl->ErrorCode();
+        }
+
+        CURVEFS_ERROR retcode = CURVEFS_ERROR::FAILED;
+        FSStatusCode stcode = response.statuscode();
+        FSStatusCode2CurveFSErr(stcode, &retcode);
+        LOG_IF(WARNING, retcode != CURVEFS_ERROR::OK)
+            << "DeleteFs: fsname = " << fsName << ", errcode = " << retcode
+            << ", errmsg = " << FSStatusCode_Name(stcode);
+
+        return static_cast<int>(retcode);
+    };
+    return excutor_.DoRPCTask(task);
+}
+
+CURVEFS_ERROR MdsClientImpl::MountFs(const std::string &fsName,
+                                     const MountPoint &mountPt,
+                                     FsInfo *fsInfo) {
+    auto task = RPCTask {
+        MountFsResponse response;
+        mdsbasecli_->MountFs(fsName, mountPt, &response, cntl, channel);
+        if (cntl->Failed()) {
+            LOG(WARNING) << "MountFs Failed, errorcode = " << cntl->ErrorCode()
+                         << ", error content:" << cntl->ErrorText()
+                         << ", log id = " << cntl->log_id();
+            return -cntl->ErrorCode();
+        }
+
+        CURVEFS_ERROR retcode = CURVEFS_ERROR::FAILED;
+        FSStatusCode stcode = response.statuscode();
+        FSStatusCode2CurveFSErr(stcode, &retcode);
+        LOG_IF(WARNING, retcode != CURVEFS_ERROR::OK)
+            << "MountFs: fsname = " << fsName
+            << ", mountPt = " << mountPt.DebugString()
+            << ", errcode = " << retcode
+            << ", errmsg = " << FSStatusCode_Name(stcode);
+
+        if (response.has_fsinfo()) {
+            fsInfo->CopyFrom(response.fsinfo());
+        }
+
+        return static_cast<int>(retcode);
+    };
+    return excutor_.DoRPCTask(task);
+}
+
+CURVEFS_ERROR MdsClientImpl::UmountFs(const std::string &fsName,
+                                      const MountPoint &mountPt) {
+    auto task = RPCTask {
+        UmountFsResponse response;
+        mdsbasecli_->UmountFs(fsName, mountPt, &response, cntl, channel);
+        if (cntl->Failed()) {
+            LOG(WARNING) << "UmountFs Failed, errorcode = " << cntl->ErrorCode()
+                         << ", error content:" << cntl->ErrorText()
+                         << ", log id = " << cntl->log_id();
+            return -cntl->ErrorCode();
+        }
+
+        CURVEFS_ERROR retcode = CURVEFS_ERROR::FAILED;
+        FSStatusCode stcode = response.statuscode();
+        FSStatusCode2CurveFSErr(stcode, &retcode);
+        LOG_IF(WARNING, retcode != CURVEFS_ERROR::OK)
+            << "UmountFs: fsname = " << fsName
+            << ", mountPt = " << mountPt.DebugString()
+            << ", errcode = " << retcode
+            << ", errmsg = " << FSStatusCode_Name(stcode);
+
+        return static_cast<int>(retcode);
```

Comment:
Same as above

Reply:
curvefs/src/client/mds_client.cpp: In lambda function:
curvefs/src/client/mds_client.cpp:80:16: error: cannot convert 'curvefs::client::CURVEFS_ERROR' to 'int' in return
         return retcode;